x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/SparseFileTracker.java

```diff
@@ -255,18 +255,21 @@ private long computeLengthOfRanges() {
         }
     }
 
-    private void onGapSuccess(final long start, final long end) {
+    private boolean assertPendingRangeExists(Range range) {
+        assert Thread.holdsLock(mutex);
+        final SortedSet<Range> existingRanges = ranges.tailSet(range);
+        assert existingRanges.isEmpty() == false;
+        final Range existingRange = existingRanges.first();
+        assert existingRange.start == range.start && existingRange.end == range.end && existingRange.isPending();
+        return true;
+    }
+
+    private void onGapSuccess(final Range existingRange) {
```

Comment:
Naming suggestion (here and elsewhere):

```suggestion
    private void onGapSuccess(final Range gapRange) {
```

Reply:
That's better, thanks